packages/overlay/src/vaadin-overlay-stack-mixin.js

```diff
@@ -23,6 +23,24 @@ const getAttachedInstances = () =>
  */
 const getOverlayInstances = () => getAttachedInstances().filter((el) => el.$.overlay);
 
+/**
+ * Returns true if all the instances on top of the overlay are nested overlays.
+ * @private
+ */
+const hasOnlyNestedOverlays = (overlay) => {
```

Comment:
The naming is a bit confusing, because this also implicitly checks if this is the last overlay. It looks like we need the early return if `next` is undefined. 

Maybe it would be more readable if the check in `bringToFront` would use both `isLastOverlay(this) || hasOnlyNestedOverlays(this)`, even if it's redundant..